src/vehicle.cpp

```diff
@@ -4693,7 +4773,12 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
                 }
             }
         } else if( ret.type == veh_coll_body ) {
-            int dam = obj_dmg*dmg_mod/100;
+            int dam = obj_dmg * dmg_mod / 100;
+            if( chainsaw_on && part_info( ret.part ).has_flag( "CHAINSAW" ) &&
+                parts[part].hp > 0 ) {
+                dam += 66;
```

Comment:
The effect of collision between a chainsaw and a monster should be
different than the effect from a monster entering a tile with a chainsaw.
How about one check in vehicle collision code, and another check in the
trap handling code?